File: Examples/CSharp/Geometry-and-Hierarchy/CubeScene.cs
using System;
using System.Collections.Generic;
using System.IO;
using Aspose.ThreeD;
using Aspose.ThreeD.Entities;

namespace Aspose._3D.Examples.CSharp.Geometry_Hierarchy
{
    class CubeScene
    {
        public static void Run()
        {
            // ExStart:CreateCubeScene           
            // Initialize scene object
            Scene scene = new Scene();
            
            // Initialize Node class object
            Node cubeNode = new Node("cube");

            // Call Common class create mesh using polygon builder method to set mesh instance 
            Mesh mesh = Common.CreateMeshUsingPolygonBuilder();           

            // Point node to the Mesh geometry
            cubeNode.Entity = mesh;
            
            // Add Node to a scene
            scene.RootNode.ChildNodes.Add(cubeNode);           
            
            // The path to the documents directory.
            string MyDir = RunExamples.GetDataDir();
            MyDir = MyDir + RunExamples.GetOutputFilePath("CubeScene.fbx");          

            // Save 3D scene in the supported file formats
            scene.Save(MyDir, FileFormat.FBX7400ASCII);           
            // ExEnd:CreateCubeScene

            Console.WriteLine("\nCube Scene created successfully.\nFile saved at " + MyDir);

        }
    }
}

File: Examples/VisualBasic/Geometry-and-Hierarchy/CubeScene.vb
'///////////////////////////////////////////////////////////////////////
' Copyright 2015 Aspose Pty Ltd. All Rights Reserved.
'
' This file is part of Aspose.3D. The source code in this file
' is only intended as a supplement to the documentation, and is provided
' "as is", without warranty of any kind, either expressed or implied.
'///////////////////////////////////////////////////////////////////////
Imports System.IO
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Entities

Namespace Geometry_Hierarchy
    Public Class CubeScene
        Public Shared Sub Run()

            ' ExStart:CreateCubeScene    
            ' Initialize scene object
            Dim scene As New Scene()

            ' Initialize Node class object
            Dim cubeNode As New Node("cube")

            ' Call Common class create mesh using polygon builder method to set mesh instance 
            Dim mesh As Mesh = Common.CreateMeshUsingPolygonBuilder()

            ' Point node to the Mesh geometry
            cubeNode.Entity = mesh

            ' Add Node to a scene
            scene.RootNode.ChildNodes.Add(cubeNode)

            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()
            MyDir = MyDir & RunExamples.GetOutputFilePath("CubeScene.fbx")
            
            ' Save 3D scene in the supported file formats
            scene.Save(MyDir, FileFormat.FBX7400ASCII)
            ' ExEnd:CreateCubeScene    

            Console.WriteLine(Convert.ToString(vbLf & "Cube Scene created successfully." & vbLf & "File saved at ") & MyDir)

        End Sub
    End Class
End Namespace